src/device-manager/cocoa/NLGenericTraitUpdatableDataSink.h

```diff
@@ -0,0 +1,277 @@
+/*
+ *
+ *    Copyright (c) 2020 Google LLC.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines NLGenericUpdatableDataSink interface
+ *
+ */
+
+#import <Foundation/Foundation.h>
+#import "NLWeaveErrorCodes.h"
+
+typedef void (^GenericTraitUpdatableDataSinkCompletionBlock)(id owner, id data);
+typedef void (^GenericTraitUpdatableDataSinkFailureBlock)(id owner, NSError * error);
+
+@interface NLGenericTraitUpdatableDataSink : NSObject
+@property (copy, readonly) NSString * name;
+@property (readonly) dispatch_queue_t resultCallbackQueue;
+@property (weak) id owner;
+
+/**
+ *  @brief Disable default initializer inherited from NSObject
+ */
+- (instancetype)init NS_UNAVAILABLE;
+
+/**
+ * convert weave error to string
+ */
+- (NSString *)toErrorString:(WEAVE_ERROR)err;
```

Comment:
nit: Generally this would be named to start with a noun because it returns something and doesn't have side effects. 

```
- (NSString *)stringDescribingError:(WEAVE_ERROR)err;
```

Feel free to ignore my suggestion if -toXX is the common pattern in the WeaveDeviceManager.

Reply:
yes, it is currently the common pattern from WeaveDeviceManager